src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray`1+Builder.cs

```diff
@@ -420,14 +411,9 @@ public bool Contains(T item)
             /// </summary>
             public T[] ToArray()
             {
-                var tmp = new T[this.Count];
-                var elements = _elements;
-                for (int i = 0; i < tmp.Length; i++)
-                {
-                    tmp[i] = elements[i];
-                }
-
-                return tmp;
+                T[] result = new T[_count];
```

Comment:
I think we try to reuse our own properties rather than fields when available. Can you change this `_count` to `this.Count` as it was before?